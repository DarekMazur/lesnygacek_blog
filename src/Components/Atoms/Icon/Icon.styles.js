import styled from 'styled-components';

export const StyledIcon = styled.span`
  display: inline-block;
  height: ${({ size }) => (size === 'medium' ? `4rem` : size === 'small' ? `3rem` : `5rem`)};
  color: ${({ theme, light }) => (light ? theme.colors.white : theme.colors.dark)};
  padding: 0.5rem;
  transition: color 200ms;

  svg {
    height: 100%;
    width: auto !important;
  }

  &:hover {
    color: ${({ theme, light }) => (light ? theme.colors.brown : theme.colors.lightGreen)};
  }
`;
