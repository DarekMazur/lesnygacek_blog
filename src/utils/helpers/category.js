export const categoryIcon = {
  tools: 'axe',
  trips: ['fas', 'hiking'],
  thoughts: 'campfire',
  others: 'backpack',
};

export const categoryTitle = {
  tools: 'Narzędzia',
  trips: 'Wyprawy',
  thoughts: 'Przemyślenia',
  others: 'Inne',
};
