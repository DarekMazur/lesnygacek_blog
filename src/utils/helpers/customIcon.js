import React from 'react';

export const customIcon = {
  backpack: (
    <svg width="46" height="52" viewBox="0 0 46 52" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M7.01922e-05 21.1556C0.000159205 16.2422 5.47612 9.69104 10.131 9.69104H13.8274V4.50468C13.8274 1.9017 16.145 0.00077452 18.756 0.000745851C21.7679 0.000712781 23.5955 0.000751969 26.6965 0.000745851C29.4346 0.00074045 31.8988 2.04795 31.8988 4.50468V9.69104H35.5953C39.7023 9.69104 46 15.9692 46 21.1556V45.859C46 49.3272 42.9681 51.2784 39.5655 52.0007V35.4863C39.5655 31.9057 36.5858 29.0716 32.9941 29.0716H13.006C9.37592 29.0716 6.29768 31.8674 6.29768 35.4863V52.0007C3.04243 51.2999 6.99369e-05 49.5885 7.01922e-05 46.2685C7.08114e-05 38.2159 -8.80495e-05 29.8905 7.01922e-05 21.1556ZM14.2381 21.1556C12.9015 21.1556 12.9015 17.7435 14.2381 17.7435H31.3512C33.2677 17.7435 33.1308 21.1556 31.3512 21.1556H14.2381ZM26.6965 9.69104H19.0297C18.8785 9.69104 18.756 9.56875 18.756 9.41802V5.5965C18.756 5.06355 19.0427 4.50468 19.5773 4.50468H26.2856C26.79 4.50468 26.9701 5.09368 26.9701 5.5965V9.41802C26.9701 9.56875 26.8477 9.69104 26.6965 9.69104Z"
        fill="currentColor"
      />
      <path
        d="M9.8572 35.0769V38.7619H35.8691V35.0769C35.8691 33.6026 34.4728 32.6202 32.9941 32.6202C26.0118 32.6202 19.8511 32.3472 13.006 32.3472C11.3764 32.3472 9.8572 33.4523 9.8572 35.0769Z"
        fill="currentColor"
      />
      <path d="M9.8572 42.4469H35.8691V52.0007H9.8572V42.4469Z" fill="currentColor" />
    </svg>
  ),
  campfire: (
    <svg width="47" height="47" viewBox="0 0 47 47" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M1.4841 31.8247C-1.34578 30.596 0.253651 25.8436 3.08359 26.6641C10.7118 28.8757 23.6308 33.7906 23.6308 33.7906C23.6308 33.7906 35.5654 29.1216 43.4398 26.6641C47.0078 25.5505 48.2382 30.8564 45.5314 31.8247C37.2878 34.7736 32.4895 36.7396 32.4895 36.7396C32.4895 36.7396 36.9188 38.4598 45.5314 41.7773C48.2382 42.8199 46.6387 48.0438 43.4398 46.815C36.1398 44.0109 23.6308 39.8113 23.6308 39.8113C23.6308 39.8113 8.49718 44.972 3.08359 46.815C0.376681 47.7366 -1.09976 42.8094 1.4841 41.7773C6.4056 39.8113 14.6491 36.7396 14.6491 36.7396C14.6491 36.7396 6.87301 34.1644 1.4841 31.8247Z"
        fill="currentColor"
      />
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M10.5888 18.6774C11.7387 25.1258 17.0723 29.2867 23.6308 29.2444C30.2134 29.2019 35.3546 24.8963 36.5496 18.4317C38.0261 10.4451 29.9056 2.21266 28.9213 2.21266C28.1831 2.21266 25.9684 4.91584 25.9684 4.91584C25.9684 4.91584 21.6621 0.00103929 20.4318 0.000976563C19.2014 0.000913836 8.98929 9.70783 10.5888 18.6774ZM16.8637 18.3088C16.1255 15.1492 20.1857 10.6908 20.6778 10.6908C21.17 10.6908 25.9684 16.9572 25.9684 16.9572C25.9684 16.9572 28.3061 13.7626 28.9213 13.7626C29.1674 13.7626 30.5067 16.5476 30.2747 18.4317C29.8663 21.7491 26.9774 24.0067 23.6308 23.9609C18.4632 23.8901 17.6371 21.619 16.8637 18.3088Z"
        fill="currentColor"
      />
    </svg>
  ),
  axe: (
    <svg width="66" height="53" viewBox="0 0 66 53" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M24.8479 21.4431L31.6062 28.084C31.6062 28.084 11.4693 48.8367 6.91779 52.8489C6.50402 53.1256 4.9664 53.0197 4.29724 52.5722C2.64216 51.4654 1.12501 50.3586 0.297454 48.56C-0.140286 47.6989 -0.132163 46.8074 0.573282 45.9313C3.35832 42.4725 24.8479 21.4431 24.8479 21.4431Z"
        fill="currentColor"
      />
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M23.4686 14.6639C23.0548 14.2328 23.0657 12.5886 23.4686 12.1735C28.5718 6.91618 31.4682 4.14918 35.0542 0.413657C35.4952 -0.0457619 37.123 -0.220511 37.8126 0.413657C39.4677 1.93562 40.0194 2.489 41.3986 3.87245C41.6653 4.13987 42.1136 4.15511 42.3641 3.87245C43.4675 2.62736 44.2761 1.81227 45.6743 0.69036C46.3639 0.137003 47.3605 -0.0189316 48.2948 0.413657C50.0878 1.24382 51.0533 2.48894 52.4325 4.0108C53.0266 4.6663 53.1221 6.29359 52.4325 7.05453C51.0533 8.57641 50.2257 9.40652 48.9844 10.79C48.7682 11.031 48.7495 11.3975 48.9844 11.6201C51.4671 13.9721 54.2255 16.7391 54.2255 16.7391H65.8111C65.8111 16.7391 67.8744 28.084 58.3632 37.0768C51.1574 43.89 40.0194 42.7492 40.0194 42.7492L39.7436 31.266C39.7436 31.266 30.6406 22.1349 23.4686 14.6639ZM28.9855 13.0036C28.7114 13.2633 28.7251 13.6987 28.9855 13.9721C34.9162 20.1979 44.5709 29.4675 44.5709 29.4675V38.0453C44.5709 38.0453 48.2948 39.0137 54.9151 33.8947C61.1217 28.3607 61.1217 21.5814 61.1217 21.5814L52.2946 21.3047C52.2946 21.3047 42.3641 11.6201 36.7092 5.80937C36.5211 5.61606 36.2198 5.62865 36.0196 5.80937C33.2612 8.29971 31.4682 10.6517 28.9855 13.0036Z"
        fill="currentColor"
      />
    </svg>
  ),
};
