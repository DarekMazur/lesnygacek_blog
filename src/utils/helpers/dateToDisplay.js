export const dateToDisplay = (date) => {
  const Day = new Date(date).getDate() < 10 ? `0${new Date(date).getDate()}` : new Date(date).getDate();
  const Month = new Date(date).getMonth() + 1 < 10 ? `0${new Date(date).getMonth() + 1}` : new Date(date).getMonth() + 1;
  const Year = new Date(date).getFullYear();

  const propperDate = `${Day}.${Month}.${Year}`;
  return propperDate;
};
