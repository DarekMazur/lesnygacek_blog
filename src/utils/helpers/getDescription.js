export const getDescription = (description, content) => {
  if (description) {
    return description;
  } else {
    const generatedDescription = content.split('').slice(0, 170).join('');
    return `${generatedDescription}[...]`;
  }
};
