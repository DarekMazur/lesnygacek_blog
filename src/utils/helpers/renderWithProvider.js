import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../themes/theme';
import { GlobalStyle } from '../../styles/globalStyle';

export const renderWithProvider = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
