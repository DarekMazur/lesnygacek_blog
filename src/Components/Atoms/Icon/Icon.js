import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { StyledIcon } from './Icon.styles';
import { customIcon } from '../../../utils/helpers/customIcon';

const Icon = ({ icon, iconType, light, size }) => {
  return (
    <StyledIcon light={light} size={size} data-testid={iconType}>
      {icon && icon === 'fa' ? <FontAwesomeIcon data-testid={icon} icon={iconType} /> : customIcon[iconType]}
    </StyledIcon>
  );
};

export default Icon;
