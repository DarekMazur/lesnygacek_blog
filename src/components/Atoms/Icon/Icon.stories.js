import Icon from './Icon';

export default {
  title: 'Components/Atoms/Icon',
  component: Icon,
};

const Template = (args) => <Icon {...args} />;

export const FontAwesomeDefault = Template.bind({});
FontAwesomeDefault.args = {
  icon: 'fa',
  iconType: ['fab', 'facebook-messenger'],
  size: 'default',
};

export const FontAwesomeLight = Template.bind({});
FontAwesomeLight.args = {
  icon: 'fa',
  iconType: ['fab', 'facebook-messenger'],
  size: 'default',
  light: true,
};

export const CustomIconDefault = Template.bind({});
CustomIconDefault.args = {
  iconType: 'axe',
  size: 'default',
};

export const IconMedium = Template.bind({});
IconMedium.args = {
  iconType: 'axe',
  size: 'medium',
};

export const IconSmall = Template.bind({});
IconSmall.args = {
  iconType: 'axe',
  size: 'small',
};

export const CustomIconLight = Template.bind({});
CustomIconLight.args = {
  iconType: 'axe',
  size: 'default',
  light: true,
};

export const IconMediumLight = Template.bind({});
IconMediumLight.args = {
  iconType: 'axe',
  size: 'medium',
  light: true,
};

export const IconSmallLight = Template.bind({});
IconSmallLight.args = {
  iconType: 'axe',
  size: 'small',
  light: true,
};
