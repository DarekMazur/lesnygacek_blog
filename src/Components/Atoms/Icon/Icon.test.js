import React from 'react';
import { renderWithProvider } from '../../../helpers/renderWithProvider';
import Icon from './Icon';

const container = renderWithProvider(<Icon icon={'fa'} iconType={['fab', 'facebook-messenger']} light size="default" />);
// const customContainer = renderWithProvider(<Icon iconType="axe" size="default" />);

describe('Render FA Icon:', () => {
  it('- rendered component match to snapshot:', () => {
    renderWithProvider(<Icon icon={'fa'} iconType={['fab', 'facebook-messenger']} light size="default" />);
    expect(container).toMatchSnapshot();
  });
});
