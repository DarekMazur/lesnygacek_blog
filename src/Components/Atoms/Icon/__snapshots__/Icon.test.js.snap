// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FA Icon: - rendered component match to snapshot: 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bczRLJ gjOKni"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-facebook-messenger fa-w-16 "
          data-icon="facebook-messenger"
          data-prefix="fab"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256.55 8C116.52 8 8 110.34 8 248.57c0 72.3 29.71 134.78 78.07 177.94 8.35 7.51 6.63 11.86 8.05 58.23A19.92 19.92 0 0 0 122 502.31c52.91-23.3 53.59-25.14 62.56-22.7C337.85 521.8 504 423.7 504 248.57 504 110.34 396.59 8 256.55 8zm149.24 185.13l-73 115.57a37.37 37.37 0 0 1-53.91 9.93l-58.08-43.47a15 15 0 0 0-18 0l-78.37 59.44c-10.46 7.93-24.16-4.6-17.11-15.67l73-115.57a37.36 37.36 0 0 1 53.91-9.93l58.06 43.46a15 15 0 0 0 18 0l78.41-59.38c10.44-7.98 24.14 4.54 17.09 15.62z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
    <div>
      <span
        class="sc-bczRLJ gjOKni"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-facebook-messenger fa-w-16 "
          data-icon="facebook-messenger"
          data-prefix="fab"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256.55 8C116.52 8 8 110.34 8 248.57c0 72.3 29.71 134.78 78.07 177.94 8.35 7.51 6.63 11.86 8.05 58.23A19.92 19.92 0 0 0 122 502.31c52.91-23.3 53.59-25.14 62.56-22.7C337.85 521.8 504 423.7 504 248.57 504 110.34 396.59 8 256.55 8zm149.24 185.13l-73 115.57a37.37 37.37 0 0 1-53.91 9.93l-58.08-43.47a15 15 0 0 0-18 0l-78.37 59.44c-10.46 7.93-24.16-4.6-17.11-15.67l73-115.57a37.36 37.36 0 0 1 53.91-9.93l58.06 43.46a15 15 0 0 0 18 0l78.41-59.38c10.44-7.98 24.14 4.54 17.09 15.62z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bczRLJ gjOKni"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-facebook-messenger fa-w-16 "
        data-icon="facebook-messenger"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256.55 8C116.52 8 8 110.34 8 248.57c0 72.3 29.71 134.78 78.07 177.94 8.35 7.51 6.63 11.86 8.05 58.23A19.92 19.92 0 0 0 122 502.31c52.91-23.3 53.59-25.14 62.56-22.7C337.85 521.8 504 423.7 504 248.57 504 110.34 396.59 8 256.55 8zm149.24 185.13l-73 115.57a37.37 37.37 0 0 1-53.91 9.93l-58.08-43.47a15 15 0 0 0-18 0l-78.37 59.44c-10.46 7.93-24.16-4.6-17.11-15.67l73-115.57a37.36 37.36 0 0 1 53.91-9.93l58.06 43.46a15 15 0 0 0 18 0l78.41-59.38c10.44-7.98 24.14 4.54 17.09 15.62z"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
